[KED-2530] Add details about testing approach in style guide (#426)
quantumblacklabs/kedro-viz: STYLE_GUIDE.md

We use [Jest](https://jestjs.io/), [Enzyme](https://enzymejs.github.io/enzyme/), and [Testing Library](https://testing-library.com/) for JavaScript testing. Most of the older tests are written with Enzyme, but we are beginning to write more tests using Testing Library as it is more flexible for testing certain browser APIs. Either is acceptable. To help with mocking the Redux store, Enzyme helper utilities can be found in `/src/utils/state.mock.js`.

> Before requesting a review on a PR, be sure to first fully self-review [by comparing changes](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-comparing-branches-in-pull-requests) line by line and make any amends with this style guide in mind.

Reviewer: ```suggestion
Before you request a review on a PR, you should first be sure to review it yourself [by comparing changes](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-comparing-branches-in-pull-requests) line by line. Make your amends with this style guide in mind.
```

> Manually [cross browser](#browser-and-device-support) and [regression test](https://en.wikipedia.org/wiki/Regression_testing) the PR branch against the behaviour of the [previous release](https://quantumblacklabs.github.io/kedro-viz/). Resolve problems where practical or otherwise make note of them in the PR text before requesting a review.

Reviewer: ```suggestion
You should manually test your PR branch [across different browsers](#browser-and-device-support) and for [regressions](https://en.wikipedia.org/wiki/Regression_testing) against the behaviour of the [previous release](https://quantumblacklabs.github.io/kedro-viz/). Resolve problems where practical or otherwise note them in the PR text before you request a review.
```
